core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/UpgradeService.java

```diff
@@ -129,6 +140,47 @@ private UpgradeOptionV4Response getUpgradeOption(Stack stack, Long workspaceId,
         }
     }
 
+    private Predicate<com.sequenceiq.cloudbreak.cloud.model.catalog.Image> getImageFilter(Image image, boolean baseImage, Stack stack) {
+        if (baseImage) {
+            return packageVersionFilter(image.getPackageVersions());
+        } else {
+            return packageVersionFilter(image.getPackageVersions())
+                    .and(parcelFilter(stack));
+        }
+    }
+
+    private Predicate<com.sequenceiq.cloudbreak.cloud.model.catalog.Image> parcelFilter(Stack stack) {
+        Set<String> parcels = clusterComponentConfigProvider.getClouderaManagerProductDetails(stack.getCluster().getId())
+                .stream()
+                .map(ClouderaManagerProduct::getParcel)
+                .collect(toSet());
+        return imageFromCatalaog -> {
+            if (imageFromCatalaog.getPreWarmParcels() == null) {
+                return false;
+            } else {
+                Set<String> imageParcels = imageFromCatalaog.getPreWarmParcels()
+                        .stream()
+                        .map(parts -> parts.get(1))
+                        .collect(toSet());
+                if (imageFromCatalaog.getStackDetails() != null) {
+                    imageParcels.add(imageFromCatalaog.getStackDetails().getRepo().getStack().get(imageFromCatalaog.getOsType()));
+                }
+                return imageParcels
+                        .equals(parcels);
+            }
+        };
+    }
+
+    private Predicate<com.sequenceiq.cloudbreak.cloud.model.catalog.Image> packageVersionFilter(Map<String, String> packageVersions) {
+        return image -> {
+            Map<String, String> catalogPackageVersions = new HashMap<>(image.getPackageVersions());
+            catalogPackageVersions.remove(SALT_BOOTSTRAP);
+            Map<String, String> originalPackageVersions = new HashMap<>(packageVersions);
+            originalPackageVersions.remove(SALT_BOOTSTRAP);
+            return originalPackageVersions.equals(catalogPackageVersions);
```

Comment:
afaik in case of upgrading to a baseimage, this will return false, because running stack's package versions contains cm and cdh version, but base images in image catalog contains only salt and salt-bootstrap in image catalog